### DISCUSSION TITLE
Dual/Multi-Pane Editor & Preview

### DISCUSSION THREAD

**Discussion by jhildenbiddle** [notaapp/talk#156] [Feature Requests]
I would love see nota offer either a dual- or multi-pane editor interface. Managing and navigating multiple editable views becomes an important part of the workflow when evolving from a "document editor" like Caret and Typora to a tool for working with wiki-like collections of linked pages.

* A **dual-pane** interface (as requested in #111) is the bare minimum IMHO. This is a feature I learned to appreciate very quickly when I took [logseq](https://logseq.com/) for a test drive. A [modifier]+click opens linked content in a new editor pane (typically to the right, but it doesn't have to be). Equally important is the ability to quickly dismiss this new panel via a keyboard shortcut, perhaps with something like [same modifier used with click]+ESC.

  ![CleanShot 2022-02-09 at 11 41 54@2x](https://user-images.githubusercontent.com/442527/153258796-a6e59545-f91a-4122-ad74-bd96f93eeb7d.png)

  <img width="830" alt="CleanShot 2022-02-09 at 11 42 09@2x" src="https://user-images.githubusercontent.com/442527/153258823-5210f426-4862-4d61-8310-37fea2a78feb.png">

* A **multi-pane** interface is a more flexible option that allows users to build a "stack" of open documents as they [modifier]+click links within adjacent panes. This approach is useful for navigating deeply nested or a complex graph of document relationships. [Notabase](https://notabase.io/) offers this (see attached video below), and [Andy's Working Notes](https://notes.andymatuschak.org/About_these_notes) was an inspiration for the author of Notabase. As above, the ability to easily dismiss and navigate these panes via keyboard shortcut is important. After trying Notabase, I would say this is where Notabase's implementation breaks down (no easy way to dismiss panes, navigate forward/back, focus on a selected pane, etc.)

  https://user-images.githubusercontent.com/442527/153259967-317096ea-c56d-4f22-b323-f56bc5087299.mp4

Excited to see how Nota progress. Keep up the great work!

**Comment by astoilkov** [comment]
Thank you very much for the extended explanation, image, and video. This helps a lot.

You are absolutely right. We believe in dual/multi-pane editing. We've started exploring it. There are different possible implementations and we are currently trying to extensively research the benefits of each one. We are asking questions such as "What's the best implementation on the marker?", "Is there a better implementation no one have though about?", "What are the most important parts to get right?", "Does another feature exists that can be implemented and satisfy the same use cases?". We also want to have a list of ways people use it so we can optimize for the most popular.

As you can see it's a lot of research. This means it can take some time to figure out. We can do an implementation (pick what our feelings tell us currently) without the research but this isn't part of our philosophy.

**Comment by Auratelience** [comment]
Perhaps a combination of stacks of editors (with tabs) and various panes could be a good idea.

**Comment by doctor-ew** [comment]
Take a look at how MacDown handles this: [screenshot of macdown dual pane](https://url.doctorew.com/2p9asjfy)

**Comment by claudiorrrr** [comment]
Related, on that dual panel it would be cool to be able to see the rendered preview such as you can see with Shift CMD R
